bridges/otelzap/encoder.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"time"
+
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+var _ zapcore.ObjectEncoder = (*objectEncoder)(nil)
```

Comment:
What about `zapcore.ArrayEncoder`? Separate PR?

Reply:
yes that was the idea, should I add it in this PR itself

Reply:
No. This is fine.